api.md

```diff
@@ -0,0 +1,148 @@
+## API
+
+All requests can be authenticated by a signed cookie with the session ID.
+
+### POST /session
+Start a new observation session.
+
+Request:
+```
+{
+  name: 'name',
+  methodology: {
+    observation_type: 'focal_animal',
+    focal_animal: 'tarzan',
+    interval_seconds: '100'
+  }
+
+  note: 'thing',
+
+  // extra metadata
+  location: {
+    lat, long
+  }
+}
+```
+
+Response:
+
+```
+Headers:
+X-Session-Id=YourResourceKey
+
+or
+
+{
+  session: {
+    id: your_resource_key
+  }
+}
+```
+
+### PATCH /session
+Start or finish the session.
+
+```
+{
+  session: {
+    state: 'finished' // (start|finish)
+  }
+}
+```
+
+### GET /session/schema
+Get the active schema for this observation session.
+```
+{
+  subjects: [
+    { name: 'tarzan', groups: ['humans'] },
+    { name: 'jane', groups: [] }
+  ],
+  behaviors: [
+    {
+      name: 'walking',
+      type: 'state'
+      group: 'posture',
+      target: 'none',
+      exclusive: false,
+      modifiers: ['peaceful', 'agitated']
+    },
+    {
+      name: 'standing',
+      type: 'state'
+      group: 'posture',
+      exclusive: true,
+      modifiers: ['peaceful', 'agitated']
+    },
+    {
+      name: 'pointing',
+      type: 'event',
+      group: 'actions',
+      modifiers: []
+    }
+  ]
+}
+```
+
+### POST/PUT /session/schema
+Create a new observation schema.
+
+TODO: What should creating subject groups look like?
+```
+{
+	schema: {
+		subjects: [
+			'tarzan', 'jane'
+		],
+		
+		behaviors: [
```

Comment:
we need a field for whether receivers are allowed - e.g. "walking" shouldn't allow one, "grooming" should.